### ISSUE TITLE
SelectEditor.getValue - Cannot read property 'getValue' of null

### ISSUE THREAD

**Issue by simoncodearcs** [nhn/tui.grid#1217]
Set up the grid with a built in editor of type: 'select' on one or more columns.  

The data we have on row 1 for example is:

```
[
	{
		Address 1: "1223 Woodvale",
		City: "Capital City",
		Country: "USA",
		Date Of Birth: "1/1/20",
		Email/Username: "some.email@wherever.com",
		First Name: "John",
		Grade Level: "4",
		Instructor Email: "another.email@wherever.com",
		Last Name: "Brown",
		Parent Email: "onemore.email@wherever.com",
		Phone: "123-444-2324",
		State/Province: "MI",
		Timezone: "EST",
		Zipcode: "49686"
	}
]
```

For timezone, the options we have are:

```
[
	{text: "(UTC-10:00) Adak", value: "America/Adak"}
	{text: "(UTC-10:00) Honolulu", value: "Pacific/Honolulu"}
	{text: "(UTC-09:00) Anchorage", value: "America/Anchorage"}
	{text: "(UTC-09:00) Juneau", value: "America/Juneau"}
	{text: "(UTC-09:00) Metlakatla", value: "America/Metlakatla"}
	{text: "(UTC-09:00) Nome", value: "America/Nome"}
	{text: "(UTC-09:00) Sitka", value: "America/Sitka"}
	{text: "(UTC-09:00) Yakutat", value: "America/Yakutat"}
	{text: "(UTC-08:00) Los Angeles", value: "America/Los_Angeles"}
	{text: "(UTC-07:00) Boise", value: "America/Boise"}
	{text: "(UTC-07:00) Denver", value: "America/Denver"}
	{text: "(UTC-07:00) Phoenix", value: "America/Phoenix"}
	{text: "(UTC-06:00) Chicago", value: "America/Chicago"}
	{text: "(UTC-06:00) Indiana/Knox", value: "America/Indiana/Knox"}
	{text: "(UTC-06:00) Indiana/Tell City", value: "America/Indiana/Tell_City"}
	{text: "(UTC-06:00) Menominee", value: "America/Menominee"}
	{text: "(UTC-06:00) North Dakota/Beulah", value: "America/North_Dakota/Beulah"}
	{text: "(UTC-06:00) North Dakota/Center", value: "America/North_Dakota/Center"}
	{text: "(UTC-06:00) North Dakota/New Salem", value: "America/North_Dakota/New_Salem"}
	{text: "(UTC-05:00) Detroit", value: "America/Detroit"}
	{text: "(UTC-05:00) Indiana/Indianapolis", value: "America/Indiana/Indianapolis"}
	{text: "(UTC-05:00) Indiana/Marengo", value: "America/Indiana/Marengo"}
	{text: "(UTC-05:00) Indiana/Petersburg", value: "America/Indiana/Petersburg"}
	{text: "(UTC-05:00) Indiana/Vevay", value: "America/Indiana/Vevay"}
	{text: "(UTC-05:00) Indiana/Vincennes", value: "America/Indiana/Vincennes"}
	{text: "(UTC-05:00) Indiana/Winamac", value: "America/Indiana/Winamac"}
	{text: "(UTC-05:00) Kentucky/Louisville", value: "America/Kentucky/Louisville"}
	{text: "(UTC-05:00) Kentucky/Monticello", value: "America/Kentucky/Monticello"}
]
```

**To Reproduce**

1. Ensure the data loaded into the cells with the select editor does not contain data which match a select option
2. Click to edit one of the cells which has a select for its editor
3. Without selecting an option, click elsewhere on the grid

**Observed behaviour**
The grid becomes unresponsive to clicks, and you cannot click other cells on the grid to edit them, or select an option from the cell currently in edit mode. 
An unhandled error appears in the console.

**Expected behavior**
The select should close, and the unedited original cell data should be shown. I should be able to select another cell on the grid. 

**Screenshots**
![tui-grid-selecteditor-error](https://user-images.githubusercontent.com/58743379/98012176-b28eed80-1df0-11eb-94a5-c02acb7a23a0.gif)

![tui-grid-SelectEditor-error-breakpoint](https://user-images.githubusercontent.com/58743379/98011790-398f9600-1df0-11eb-8abd-3d16351408b9.png)

**Desktop**
 - OS: macOS
 - Browser: Chrome
 - Version: 86

**Comment by js87zz** [comment]
@simoncodearcs 
Can you upload your example in the codepen or jsfiddle?

**Comment by simoncodearcs** [comment]
Hi @js87zz, yes, here is an example on Codepen where you can see the issue: https://codepen.io/simoncodearcs/pen/YzWvBpo

1. double click one of the cells in the timezone column
2. don't select a timezone, instead, click on one of the cells in another column

If you have the browser devtools open, you will see the Uncaught TypeError: Cannot read property 'getValue' of null error logged. 

![image](https://user-images.githubusercontent.com/58743379/98096207-59bc6500-1e83-11eb-96c1-5c83ad9b6d96.png)

**Comment by Garden92** [comment]
I have the same problem.

@simoncodearcs 
Did you apply a grid filter?

There's nothing wrong with the select box.
However, applying the filter results in an error

in my case, (example24-filter.html)
```javascript
...
<script type="text/babel" class="code-js">
    const grid = new tui.Grid({
      el: document.getElementById('grid'),
      data: gridData,
      bodyHeight: 500,
      columns: [
        {
          header: 'Name',
          name: 'name',
          filter: { type: 'text', showApplyBtn: true, showClearBtn: true }
        },
        {
          header: 'Artist',
          name: 'artist',
          filter: 'select'
        },
        {
          header: 'Price',
          name: 'price',
          filter: 'number'
        },
        {
          header: 'Release',
          name: 'release',
          filter: {
            type: 'date',
            format: 'yyyy.MM.dd'
          }
        },
        {
          header: 'Genre',
          name: 'genre',
          filter: {
            type: 'select'
          }
        }, 
        {
          header: 'Type',
          name: 'typeCode',
          formatter: 'listItemText',
          editor: {
            type: 'select',
            options: {
              listItems: [
                { text: 'Deluxe', value: '1' },
                { text: 'EP', value: '2' },
                { text: 'Single', value: '3' }
              ]
            }
          }
        }

      ]
    });
  </script>
```

> Init 
<img width="1671" alt="스크린샷 2020-11-07 20 51 12" src="https://user-images.githubusercontent.com/59313582/98440253-302d6300-213b-11eb-823a-d903eb91ec05.png">

> Change SelectBox - (Clear when no filter applied)
<img width="1680" alt="스크린샷 2020-11-07 20 54 43" src="https://user-images.githubusercontent.com/59313582/98440337-87333800-213b-11eb-9aae-aff86353f454.png">

> Change SelectBox - (Error when applying filter)
<img width="1680" alt="스크린샷 2020-11-07 20 56 19" src="https://user-images.githubusercontent.com/59313582/98440420-e133fd80-213b-11eb-8ba6-5d5d8e108eb6.png">


Source analysis result confirmed that LayerComp (EditingLayerComp) finishEditing function is called twice when applying filter.

> my solution(tui-grid.js)
```javascript

EditingLayerComp.prototype.finishEditing = function (save) {

        var _a = this.props, dispatch = _a.dispatch, editingAddress = _a.editingAddress, active = _a.active;
        if (this.editor && active && _a.isCreate) {
            var _b = editingAddress, rowKey = _b.rowKey, columnName = _b.columnName;
            var value = this.editor.getValue();
            if (save) {
                dispatch('setValue', rowKey, columnName, value);
            }
            if (common_1.isFunction(this.editor.beforeDestroy)) {
                this.editor.beforeDestroy();
            }
            dispatch('finishEditing', rowKey, columnName, value);
            _a.isCreate = false;
        }
    };

EditingLayerComp.prototype.createEditor = function () {
        var _a = this.props, allColumnMap = _a.allColumnMap, filteredViewData = _a.filteredViewData, editingAddress = _a.editingAddress, grid = _a.grid, cellPosRect = _a.cellPosRect; 
        _a.isCreate = ture; // add isCreate
        var _b = editingAddress, rowKey = _b.rowKey, columnName = _b.columnName;
        var _c = cellPosRect, right = _c.right, left = _c.left;
        var columnInfo = allColumnMap[columnName];
        var _d = common_1.findProp('rowKey', rowKey, filteredViewData).valueMap[columnName], value = _d.value, formattedValue = _d.formattedValue;
        var EditorClass = columnInfo.editor.type;
        var editorProps = {
            grid: grid,
            rowKey: rowKey,
            columnInfo: columnInfo,
            value: value,
            formattedValue: formattedValue,
            width: right - left,
            portalEditingKeydown: this.handleKeyDown,
        };
        var cellEditor = new EditorClass(editorProps);
        var editorEl = cellEditor.getElement();
        if (editorEl && this.contentEl) {
            this.contentEl.appendChild(editorEl);
            this.editor = cellEditor;
            if (common_1.isFunction(cellEditor.mounted)) {
                // To access the actual mounted DOM elements
                setTimeout(function () {
                    cellEditor.mounted();
                });
            }
        }
    };
```

No more stop occurs when tui-grid.js are modified. But this is not a complete solution.

Please correct the finishEditing function so that it is not called twice.

**Comment by simoncodearcs** [comment]
Hi @Garden92, no I haven't applied any filters. It looks like slightly different error you're getting? I see `Cannot read property 'GetSelectedItem' of null` in your screenshots.

**Comment by James-dalsoft** [comment]
I also have the same problem.
No filter, version : 4.15.0

```
Uncaught TypeError: Cannot read property 'getValue' of null
    at SelectEditor.getValue (tui-grid.js:10147)
    at EditingLayerComp.finishEditing (tui-grid.js:18035)
    at EditingLayerComp.componentWillReceiveProps (tui-grid.js:18088)
    at setComponentProps (tui-grid.js:1425)
    at renderComponent (tui-grid.js:1512)
    at class_1.forceUpdate (tui-grid.js:1668)
    at class_1.setStateUsingSelector (tui-grid.js:1720)
    at Object.fn (tui-grid.js:1728)
    at callObserver (tui-grid.js:1799)
    at flush (tui-grid.js:1807)
    at batchUpdate (tui-grid.js:1785)
    at run (tui-grid.js:1817)
    at tui-grid.js:1827
    at Array.forEach (<anonymous>)
    at setValue (tui-grid.js:1826)
    at Object.set [as columnName] (tui-grid.js:1884)
```

**Comment by js87zz** [comment]
@simoncodearcs 
I think it's not the bug.
In your codepen code, the `timezone` of data has `PST` or `EST` as below.
```js
// ...
  data: [
  	{
      "First Name": "John",
      "Last Name": "Brown",
      "Timezone": "EST"
    },
    {
      "First Name": "Evan",
      "Last Name": "Browne",
      "Timezone": "PST"
    }
  ]
```
However, `PST`, `EST` are not defined in your `listItems` and It will be problem. To resolve the issue, all data of `timezone` should be defined in your `listItems`.

```js
// ...
listItems: [
  {text: "EST", value: "EST"},
  {text: "PST", value: "PST"},
  {text: "(UTC-10:00) Adak", value: "America/Adak"},
  {text: "(UTC-10:00) Honolulu", value: "Pacific/Honolulu"},
  {text: "(UTC-09:00) Anchorage", value: "America/Anchorage"},
  {text: "(UTC-09:00) Juneau", value: "America/Juneau"},
  {text: "(UTC-09:00) Metlakatla", value: "America/Metlakatla"},
  {text: "(UTC-09:00) Nome", value: "America/Nome"},
  {text: "(UTC-09:00) Sitka", value: "America/Sitka"},
  {text: "(UTC-09:00) Yakutat", value: "America/Yakutat"},
  {text: "(UTC-08:00) Los Angeles", value: "America/Los_Angeles"},
  {text: "(UTC-07:00) Boise", value: "America/Boise"},
  {text: "(UTC-07:00) Denver", value: "America/Denver"},
  {text: "(UTC-07:00) Phoenix", value: "America/Phoenix"},
  {text: "(UTC-06:00) Chicago", value: "America/Chicago"},
  {text: "(UTC-06:00) Indiana/Knox", value: "America/Indiana/Knox"},
  {text: "(UTC-06:00) Indiana/Tell City", value: "America/Indiana/Tell_City"},
  {text: "(UTC-06:00) Menominee", value: "America/Menominee"},
  {text: "(UTC-06:00) North Dakota/Beulah", value: "America/North_Dakota/Beulah"},
  {text: "(UTC-06:00) North Dakota/New Salem", value: "America/North_Dakota/New_Salem"},
  {text: "(UTC-05:00) Detroit", value: "America/Detroit"},
  {text: "(UTC-05:00) Indiana/Indianapolis", value: "America/Indiana/Indianapolis"},
  {text: "(UTC-05:00) Indiana/Marengo", value: "America/Indiana/Marengo"},
  {text: "(UTC-05:00) Indiana/Petersburg", value: "America/Indiana/Petersburg"},
  {text: "(UTC-05:00) Indiana/Vevay", value: "America/Indiana/Vevay"},
  {text: "(UTC-05:00) Indiana/Vincennes", value: "America/Indiana/Vincennes"},
  {text: "(UTC-05:00) Indiana/Winamac", value: "America/Indiana/Winamac"},
  {text: "(UTC-05:00) Kentucky/Louisville", value: "America/Kentucky/Louisville"},
  {text: "(UTC-05:00) Kentucky/Monticello", value: "America/Kentucky/Monticello"}
] 
// ...
```

**Comment by simoncodearcs** [comment]
Thanks for taking the time to look into this @js87zz, the data mismatch you have identified is the key here -- we're using the grid as part of a spreadsheet import tool, and cannot guarantee the data in the timezone (or other fields where we have a select) will be present in the listItems we have defined.

**Comment by js87zz** [comment]
@simoncodearcs 
I think you want to display the **placeholder** on the select box. right?
If so, I think it's valuable option which we need to add. But the options is not supported yet and we need to develop this option.

**Comment by simoncodearcs** [comment]
Thanks @js87zz I appreciate the consideration and your willingness to discuss handling it.  
In the mean time, I wonder if you have any suggestions for a workaround I could try?

**Comment by js87zz** [comment]
@simoncodearcs 
Curretly, You can solve the problem through the [custom editor](https://github.com/nhn/tui.grid/blob/master/packages/toast-ui.grid/docs/en/custom-editor.md) to resolve the issue, but it will be annoying to you. I think that the best way is for us to solve the script error.

**Comment by js87zz** [comment]
@simoncodearcs 
This bug(script error) has been fixed in `v4.17.0`. Thanks!
